source/systems/gaea_user_guide.rst

```diff
@@ -710,6 +719,243 @@ libraries that are compatible with Intel host compilers.
     information used by the Cray compiler wrappers. This must be called on a
     file, for example ``CC -craype-verbose test.cpp``.
 
+.. _gaea-dynamic-linking:
+
+Dynamic linking
+===============
+
+All executables built on gaea are dynamically linked, and is the only linking
+method currently supported.  This means that the executable will not contain
+the libraries it depends on, but instead will use the libraries that are
+available on the system at runtime.  Dynamic linking allows for smaller
+executable sizes and the ability to share libraries between multiple
+executables, which can save disk space and memory.  However, it also means that
+the executable may not run if the required libraries are not available at
+runtime. Dynamic linking is done using the Executable and Linking Format (ELF)
+standard. The ELF format is a common standard for executable files on Unix-like
+systems, including Linux, and is used to define the structure of executable
+files, shared libraries, and object files.
+
+.. cspell: words RPATH
+
+To help ensure that the correct libraries are available at runtime, the dynamic
+loader (:program:`ld.so`, see :manpage:`ld.so(8)`) will use location
+information stored in the executable (if available, known as ``RPATH``), then
+search the directories listed in the environment variable ``LD_LIBRARY_PATH``,
+and finally the directories listed in :file:`/etc/ld.so.conf`.  The first
+instance of a specific library found in the hierarchy of locations will be
+used.
+
+To embed the location of a library within the executable's ``RPATH``, you can
+use the ``-Wl,-rpath,<path>`` option when compiling, where ``<path>`` is the
+path to the directory that contains the library.  For example, to add the
+directory ``/path/to/some/lib`` to the executable's ``RPATH``, you would run:
+
+.. cspell: words -lsomelib myprogram
+
+.. code-block:: console
+
+    $ cc -Wl,-rpath,/path/to/some/lib -L/path/to/some/lib -lsomelib myprogram.c -o myprogram
+
+If you need to specify multiple paths to ``RPATH`` you can either specify
+multiple ``-Wl,-rpath,<path>`` options, or separate the paths with a colon
+(``:``). For example:
+
+.. code-block:: console
+
+      $ cc -Wl,-rpath,/path/to/lib1:/path/to/lib2 -L/path/to/lib1 -L/path/to/lib2 -lsomelib1  -lsomelib2 myprogram.c -o myprogram
+
+This will embed both directories in the executable's ``RPATH``.
+
+On systems that use the Cray Programming Environment, the dynamic loader is
+configured to always search the directory :file:`/opt/cray/pe/lib64`.  This
+directory contains the default Cray libraries.  Because this directory is
+included in the search path, switching the module environment does not
+completely change the run-time environment.  As an example, we'll look at using
+a non-default MPI library.  First, we will build and run using the default
+``cray-mpich`` module.  The ``mpi_test.c`` code has the following:
```

Comment:
I think you can eliminate the "First, we will build..." sentence.  You say the same thing once you've introduced the C test code.